File: jcarrollonlinev4.backend/Models/ForumThreadEntries/ThreadEntryIndexItemModel.cs
using jcarrollonlinev4.backend.Models.Fora;
using jcarrollonlinev4.backend.Models.Users;
using System.ComponentModel.DataAnnotations;

namespace jcarrollonlinev4.backend.Models.ForumThreadEntries
{
    public class ThreadEntryIndexItemModel : ThreadEntriesModelBase
    {
        private ApplicationUserModel _author = new ApplicationUserModel();
        private ForaModel _forum = new ForaModel();

        public ForaModel Forum => _forum;

        [Display(Name = "Replies")]
        public int Replies { get; set; }

        [Display(Name = "Last Reply")]
        public DateTime LastReply { get; set; }

        [Display(Name = "Recs")]
        public int Recs { get; set; }

        [Display(Name = "Views")]
        public int Views { get; set; }

        [Display(Name = "Author")]
        public ApplicationUserModel Author => _author;

        [Display(Name = "Title")]
        public string? Title { get; set; }

        [Display(Name = "Created On")]
        public DateTime CreatedAt { get; set; }
    }
}

File: jcarrollonlinev4.frontend/src/models/ForumThreadEntries/ThreadEntryIndexItemModel.ts
import { ForaGetModel } from "../Fora/ForaGetModel";
import { ApplicationUserModel } from "../Users/ApplicationUserModel";

export interface ThreadEntryIndexItemModel {
  Forum: ForaGetModel;
  Replies: number;
  LastReply: Date;
  Recs: number;
  Views: number;
  Author: ApplicationUserModel;
  Title: string;
  CreatedAt: Date;
}
